warehouse/forklift/legacy.py

```diff
@@ -642,6 +657,14 @@ def file_upload(request):
                 "The name {!r} is not allowed.".format(form.name.data),
             ) from None
 
+        # Also check for collisions with Python Standard Library modules.
+        if (packaging.utils.canonicalize_name(form.name.data) in
+                STDLIB_PROHIBITTED):
+            raise _exc_with_message(
+                HTTPBadRequest,
+                "The name {!r} is not allowed.".format(form.name.data),
```

Comment:
Should we have a distinct message here? Even if it's just including "because it is the name of a Python standard library module" in the existing message?